docs/sources/mimir/references/architecture/components/alertmanager.md

```diff
@@ -156,3 +156,66 @@ You must configure the [ruler]({{< relref "./ruler" >}}) with the addresses of A
+
+If this is a new Mimir installation, we recommend enabling support for UTF-8 before creating any tenant configurations. You can do this by changing [`utf8-strict-mode-enabled`]({{< relref "./../../../configure/configuration-parameters#alertmanager" >}}) to true.
+
+If this is an existing Mimir installation, there are a number of breaking changes that might affect existing tenant configurations. We recommend following the instructions below to ensure all existing tenant configurations are compatible with UTF-8 before enabling it.
+
+### What are the breaking changes?
+
+In order to support UTF-8, Alertmanager has added a new parser for label matchers (often abbreviated as matchers), which has a number of breaking changes. While most matchers will be forward-compatible, some will not.
+
+{{< admonition type="note" >}}
+If you are unfamiliar with what matchers are or how they are used in a tenant configuration, you can find more information about them in the [Prometheus Alertmanager documentation](https://prometheus.io/docs/alerting/latest/configuration).
+{{< /admonition >}}
+
+Grafana Mimir provides a number of tools to help you identify if any existing tenant configurations would be affected by these breaking changes, and migrate any affected tenant configurations in a way that is backwards compatible, does not change the behavior of existing matchers, and works even in Mimir installations that do not have UTF-8 enabled.
+
+### Identifying affected tenant configurations
+
+To identify affected tenant configurations, take the following steps:
+
+1. Make sure Mimir is running version 2.12 or newer with debug-level logging enabled.
+2. To identify any tenants configurations that are incompatible with UTF-8 (meaning the tenant configuration would fail to load and the [fallback configuration](#fallback-configuration) would be used instead), search Mimir server logs for lines containing `Alertmanager is moving to a new parser for labels and matchers, and this input is incompatible`. Each log line includes the invalid matcher from the tenant configuration and the ID of the affected tenant. For example:
+
+    ```
+    msg="Alertmanager is moving to a new parser for labels and matchers, and this input is incompatible. Alertmanager has instead parsed the input using the classic matchers parser as a fallback. To make this input compatible with the UTF-8 matchers parser please make sure all regular expressions and values are double-quoted. If you are still seeing this message please open an issue." input="foo=" origin=api err="end of input: expected label value" suggestion="foo=\"\"" user="1"
```

Comment:
```suggestion
    msg="Alertmanager is moving to a new parser for labels and matchers, and this input is incompatible. Alertmanager has instead parsed the input using the classic matchers parser as a fallback. To make this input compatible with the UTF-8 matchers parser, make sure all regular expressions and values are double-quoted. If you still see this message, open an issue." input="foo=" origin=api err="end of input: expected label value" suggestion="foo=\"\"" user="1"
```

Reply:
We try to avoid saying "please" in the docs.

Reply:
Can't change this, it's a literal error message from the software.